mousebender/simple.py

```diff
@@ -90,16 +88,39 @@ def parse_repo_index(html):
     return parser.mapping
 
 
-@dataclasses.dataclass(frozen=True)
 class ArchiveLink:
 
     """Data related to a link to an archive file."""
 
-    filename: str
-    url: str
-    requires_python: packaging.specifiers.SpecifierSet
-    hash_: Optional[Tuple[str, str]]
-    gpg_sig: Optional[bool]
+    filename = ""
```

Comment:
FYI none of these should be class attributes.

Reply:
Not sure where to take this, I blame my C-upbringing here. So add these fields to the constructor of the class and just set them inside there?

Reply:
Yep, exactly.

https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables

Reply:
But honestly, don't worry about this right now. Just switch to attrs and we can worry about that later as hand-crafting data classes has some slight subtleties we don't need to get into right now.

Reply:
> FYI none of these should be _class attributes_.

I missed the nuance there, _class_ attributes. I didn't realize that was the syntax before, thanks for pointing that out (makes me wonder how I've inadvertently messed things up elsewhere!).

I've switched it to `attrs` way of declaring instance variables using `attr.ib` and using `typing` since we are set to use 3.6+ and will skip over 3.5.